### PR TITLE
[flutter_tools] update to latest dwds API

### LINKED ISSUE #52922: [tool_crash] WipError 9: {code: -32000, message: Cannot find context with specified id}
## Command
```
flutter run -d chrome
```

## Steps to Reproduce
1. ...
2. ...
3. ...

## Logs
WipError: WipError 9: {code: -32000, message: Cannot find context with specified id}
```
#0      Modules._initializeMapping (package:dwds/src/debugging/modules.dart:181:9)
<asynchronous suspension>
#1      Modules.initialize (package:dwds/src/debugging/modules.dart:53:5)
#2      ChromeProxyService.createIsolate (package:dwds/src/services/chrome_proxy_service.dart:162:14)
#3      DevHandler._handleIsolateStart (package:dwds/src/handlers/dev_handler.dart:347:11)
#4      DevHandler._handleConnection.<anonymous closure> (package:dwds/src/handlers/dev_handler.dart:225:19)
#5      _rootRunUnary (dart:async/zone.dart:1141:38)
#6      _CustomZone.runUnary (dart:async/zone.dart:1034:19)
#7      _CustomZone.runUnaryGuarded (dart:async/zone.dart:936:7)
#8      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:339:11)
#9      _DelayedData.perform (dart:async/stream_impl.dart:594:14)
#10     _StreamImplEvents.handleNext (dart:async/stream_impl.dart:710:11)
#11     _PendingEvents.schedule.<anonymous closure> (dart:async/stream_impl.dart:670:7)
#12     _rootRun (dart:async/zone.dart:1129:38)
#13     _CustomZone.run (dart:async/zone.dart:1026:19)
#14     _CustomZone.runGuarded (dart:async/zone.dart:928:7)
#15     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:968:23)
#16     _rootRun (dart:async/zone.dart:1133:13)
#17     _CustomZone.run (dart:async/zone.dart:1026:19)
#18     _CustomZone.runGuarded (dart:async/zone.dart:928:7)
```
```
[✓] Flutter (Channel master, v1.16.1-pre.34, on Microsoft Windows [Version 10.0.18363.720], locale en-US)
    • Flutter version 1.16.1-pre.34 at C:\Users\rbus\Documents\flutter
    • Framework revision e6b0f5f238 (24 hours ago), 2020-03-18 21:56:02 -0400
    • Engine revision 216c420a2c
    • Dart version 2.8.0 (build 2.8.0-dev.15.0 96cf889e6b)

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at C:\Users\rbus\AppData\Local\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.3
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[✓] Android Studio (version 3.6)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 44.0.2
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)

[✓] VS Code (version 1.43.1)
    • VS Code at C:\Users\rbus\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.8.1

[✓] Connected device (2 available)
    • Chrome     • chrome     • web-javascript • Google Chrome 80.0.3987.149
    • Web Server • web-server • web-javascript • Flutter Tools

• No issues found!

```

## Flutter Application Metadata
**Version**: 1.0.0+1
**Material**: true
**Android X**: false
**Module**: false
**Plugin**: false
**Android package**: null
**iOS bundle identifier**: null

### LINKED ISSUE #53112: Default flutter create unable to open in web.
Steps to repro:

1) Checkout master branch
2) `flutter create mytest`
3) `cd mytest`
4) `flutter run -d web`
5) Open given URL.

The app doesn't load, and gives a 404 for `web_entrypoint.dart.js`

# Flutter doctor

```
[✓] Flutter (Channel unknown, v1.16.2-pre.42, on Mac OS X 10.15.3 19D76, locale
    en-US)
    • Flutter version 1.16.2-pre.42 at /Users/gspencer/code/flutter
    • Framework revision 58cad787d8 (2 hours ago), 2020-03-23 08:33:41 -0700
    • Engine revision 2d42c74a34
    • Dart version 2.8.0 (build 2.8.0-dev.16.0 ba8baa46b4)

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/gspencer/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling
      support)
    • Platform android-29, build-tools 29.0.2
    • ANDROID_HOME = /Users/gspencer/Library/Android/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3, Build version 11C29
    • CocoaPods version 1.8.4

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 38.2.3
    • Dart plugin version 191.8423
    • Java version OpenJDK Runtime Environment (build
      1.8.0_202-release-1483-b49-5587405)

[✓] IntelliJ IDEA Community Edition (version 2019.3.4)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 44.0.3
    • Dart plugin version 193.6911.25

[✓] Connected device (4 available)
    • Android SDK built for x86 • emulator-5554 • android-x86    • Android 10
      (API 29) (emulator)
    • macOS                     • macOS         • darwin-x64     • Mac OS X
      10.15.3 19D76
    • Chrome                    • chrome        • web-javascript • Google Chrome
      80.0.3987.149
    • Web Server                • web-server    • web-javascript • Flutter Tools

• No issues found!
```

I do have enable-web and enable-desktop set to true.

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Allows resolving org-dartlang-app APIs.

Possibly fixes https://github.com/flutter/flutter/issues/52922

Fixes https://github.com/flutter/flutter/issues/53112

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/ios/fallback_discovery_test.dart [hunk 1]
```diff
@@ -104,7 +104,7 @@ void main() {
       ];
     });
     when(mockVmService.getIsolate(any)).thenAnswer((Invocation invocation) async {
-      return Sentinel(kind: 'Something', valueAsString: 'Something');
+      return Sentinel(kind: 'Something', valueAsString: 'Something') as Isolate;
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/ios/fallback_discovery_test.dart]: Need to figure out the right thing to do here

File: packages/flutter_tools/lib/src/build_runner/devfs_web.dart [hunk 2]
```diff
@@ -422,12 +444,6 @@ class WebAssetServer implements AssetReader {
 
   @override
   Future<String> dartSourceContents(String serverPath) async {
-    // TODO(jonahwilliams): ensure devtools can correctly hide this file.
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/build_runner/devfs_web.dart]: FYI @grouma tested this out locally and it works great!

File: packages/flutter_tools/test/general.shard/ios/fallback_discovery_test.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/ios/fallback_discovery_test.dart]: Sentinels are thrown now, according to https://pub.dev/packages/vm_service#changelog

File: packages/flutter_tools/lib/src/build_runner/devfs_web.dart [hunk 3]
```diff
@@ -131,6 +131,22 @@ class WebAssetServer implements AssetReader {
           '.lib.js',
           (String path) async => modules,
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/build_runner/devfs_web.dart]: This is a lot of callbacks. Maybe the parameters should be named, or the closures should be hoisted and named to give some indication of what they're for.

File: packages/flutter_tools/lib/src/ios/fallback_discovery.dart [hunk 4]
```diff
@@ -159,14 +159,10 @@ class FallbackDiscovery {
         );
         final VM vm = await vmService.getVM();
         for (final IsolateRef isolateRefs in vm.isolates) {
-          final dynamic isolateResponse = await vmService.getIsolate(
+          final Isolate isolateResponse = await vmService.getIsolate(
             isolateRefs.id,
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/ios/fallback_discovery.dart]: what does getIsolate return if this id does not exist, and how should that error case be handled?

File: packages/flutter_tools/lib/src/ios/fallback_discovery.dart [hunk 5]
```diff
@@ -180,6 +176,10 @@ class FallbackDiscovery {
         // No action, we might have failed to connect.
         firstException ??= err;
         _logger.printTrace(err.toString());
+      } on Sentinel {
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/ios/fallback_discovery.dart]: Oh, I see, I guess it throws this exception. Is `Sentinel` a subtype of `Exception`?

**Review by grouma — APPROVED** [review]

File: packages/flutter_tools/lib/src/ios/fallback_discovery.dart (hunk 5 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/ios/fallback_discovery.dart]: Negative :( @bkonyi !!!!

Comment (bkonyi) [reply packages/flutter_tools/lib/src/ios/fallback_discovery.dart]: What's still throwing `Sentinel`? Everything should be throwing [SentinelException](https://pub.dev/documentation/vm_service/latest/vm_service/SentinelException-class.html) as of 4.0.0.

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/ios/fallback_discovery.dart]: The changelog in https://pub.dev/packages/vm_service#-changelog-tab- says a sentinel is thrown FYI

Comment (bkonyi) [reply packages/flutter_tools/lib/src/ios/fallback_discovery.dart]: Ah shoot, must have forgotten to update it. Originally we were going to just throw the Sentinel but then decided to make a special exception for it.

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/ios/fallback_discovery.dart]: Good call :)

File: packages/flutter_tools/lib/src/build_runner/devfs_web.dart (hunk 3 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/build_runner/devfs_web.dart]: Updated with named functions and comments explaining the purpose.

File: packages/flutter_tools/lib/src/build_runner/devfs_web.dart [hunk 6]
```diff
@@ -114,6 +114,41 @@ class WebAssetServer implements AssetReader {
...
+          return serverPath.replaceAll('.lib.js', '');
+        }
+        return null;
+      }
+      // Return the server path for modules. These are the JavaScript file names
+      // output by the frontend_server.
+      String serverPathForModule(String module) {
+        return '$module.lib.js';
+      }
+      // Return the server path for modules that have an org-dartlang-app
```
Comment (grouma) [inline packages/flutter_tools/lib/src/build_runner/devfs_web.dart]: This isn't limited to just modules. It is used for any `org-dartlang-app` resource which includes Dart files.

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/build_runner/devfs_web.dart]: Fixed

**Pull request merged**